std/format/write.d

```diff
@@ -1310,3 +1310,22 @@ void formatValue(Writer, T, Char)(auto ref Writer w, auto ref T val, scope const
     writer.formatValue(a, spec);
     assert(writer.data == "0");
 }
+
+// https://issues.dlang.org/show_bug.cgi?id=23400
+@safe pure unittest
+{
+    import std.range : nullSink;
+    import std.format.spec : singleSpec;
+
+    static struct S
+    {
+        bool opEquals(S rhs) { return false; }
```

Comment:
I'd add a comment here explaining that it's non-const on purpose.

Reply:
Comment added.